apis/v1/controllers/addresses.controller.js

```diff
@@ -73,10 +75,14 @@ exports.updateAddress = async (req, res) => {
 
 
 
-// delete address
-exports.deleteAddress = async (req, res) => {
+// soft delete address
+exports.softDeleteAddress = async (req, res) => {
     try {
-        const address = await Address.findByIdAndDelete(req.params.id);
+        const address = await Address.findByIdAndUpdate(req.params.id, {
+            status: 'INACTIVE',
+            deletedAt: Date.now(),
```

Comment:
Date.now() function returns a number representing the date. I am not sure if you will git a validation error, since as the data type of the deletedAt should be a date. I recommend however. using new Date().toISoString(). This works well in the test I have made for the user controller deleteUser function.